src/NServiceBus.Persistence.DynamoDB/Outbox/OutboxPersister.cs

```diff
@@ -237,30 +260,47 @@ IEnumerable<TransactWriteItem> Serialize(OutboxMessage outboxMessage, ContextBag
             CancellationToken cancellationToken = default)
         {
             var opsCount = context.Get<int>(OperationsCountContextProperty);
+            var currentVersion = context.Get<int>($"dynamo_version:{messageId}");
+            var nextVersion = currentVersion + 1;
 
             var now = DateTime.UtcNow;
             var expirationTime = now.Add(configuration.TimeToLive);
             int epochSeconds = AWSSDKUtils.ConvertToUnixEpochSeconds(expirationTime);
 
-            var writeRequests = new List<WriteRequest>(opsCount + 1)
+            var updateItem = new UpdateItemRequest
             {
-                new()
+                Key = new Dictionary<string, AttributeValue>
                 {
-                    PutRequest = new PutRequest
-                    {
-                        Item = new Dictionary<string, AttributeValue>
-                        {
-                            {configuration.Table.PartitionKeyName, new AttributeValue {S = $"OUTBOX#{endpointIdentifier}#{messageId}"}},
-                            {configuration.Table.SortKeyName, new AttributeValue {S = $"OUTBOX#{messageId}#0"}}, //Sort key
-                            {"TransportOperationsCount", new AttributeValue {N = "0"}},
-                            {"Dispatched", new AttributeValue {BOOL = true}},
-                            {"DispatchedAt", new AttributeValue {S = now.ToString("s")}},
-                            {configuration.Table.TimeToLiveAttributeName!, new AttributeValue {N = epochSeconds.ToString()}}
-                        },
-                    }
-                }
+                    {configuration.Table.PartitionKeyName, new AttributeValue {S = $"OUTBOX#{endpointIdentifier}#{messageId}"}},
+                    {configuration.Table.SortKeyName, new AttributeValue {S = $"OUTBOX#METADATA#{messageId}"}}, //Sort key
+                },
+                ConditionExpression = "#version = :current_version",
```

Comment:
Probably not. Although I think it doesn't hurt to use the incrementing number here. Happy to change though in case we want to adjust this specific case